src/main/java/com/xebialabs/deployit/ci/Credential.java

```diff
@@ -308,6 +311,7 @@ public static Credential fromStapler(@QueryParameter String name, @QueryParamete
 
         public FormValidation doValidateUserNamePassword(@QueryParameter String deployitServerUrl, @QueryParameter String deployitClientProxyUrl, @QueryParameter String username,
```

Comment:
Would recommend `@RequirePOST` as this seems to result in a network connection, and would therefore be a CSRF target. See https://jenkins.io/doc/developer/security/form-validation/#protecting-from-csrf

Reply:
added @RequirePOST

```
@RequirePOST
        public static Credential fromStapler(@QueryParameter String name, @QueryParameter String username, @QueryParameter Secret password,
                                             @QueryParameter String deployitServerUrl, @QueryParameter String deployitClientProxyUrl,
                                             @QueryParameter String secondaryServerUrl, @QueryParameter String secondaryProxyUrl, @QueryParameter String credentialsId, @QueryParameter boolean useGlobalCredential) {
```